C7510 compile error in debug_info.h using VS2019 on Windows

I am not sure when this issue arose, but has shown up when compiling main using VS2019 on Windows 10.  The issue is not a failure on macOS with clang. I have not checked Linux as of yet.

I am getting a build failure with a C7510 compile error (see https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-2/compiler-error-c7510?view=msvc-170) in **debug_info.h**:

```
173>D:\VSProjects\Vulkan-Samples\framework\debug_info.h(194,38): error C7510: 'Static': use of dependent template name must be prefixed with 'template'
173>D:\VSProjects\Vulkan-Samples\framework\debug_info.h(186,7): message : This diagnostic occurred in the compiler generated function 'void vkb::DebugInfo::insert(const std::string &,A &&...)'
173>Done building project "hello_triangle_1_3.vcxproj" -- FAILED.
```

The problem is with the following line:

```
	void insert(const std::string &label, A &&...args)
	{
		...
		for (auto &field : fields)
		{
			if (field->label == label)
			{
--->				if (dynamic_cast<typename field::Static<T> *>(field.get()))
				{
					field = std::make_unique<C<T>>(label, args...);
				}
				return;
			}
		}
		...

```

The docs indicate that a `template` preface is required in this circumstance, like this:

```
				if (dynamic_cast<typename field::template Static<T> *>(field.get()))
				                                 ~~~~~~~~
```

When I make this change the problem goes away and builds can proceed successfully.

--- a/framework/debug_info.h
+++ b/framework/debug_info.h
@@ -1,4 +1,4 @@
-/* Copyright (c) 2019-2024, Arm Limited and Contributors
+/* Copyright (c) 2019-2025, Arm Limited and Contributors
  *
  * SPDX-License-Identifier: Apache-2.0
  *
@@ -191,7 +191,7 @@ class DebugInfo
 		{
 			if (field->label == label)
 			{
-				if (dynamic_cast<typename field::Static<T> *>(field.get()))
+				if (dynamic_cast<typename field::template Static<T> *>(field.get()))
 				{
 					field = std::make_unique<C<T>>(label, args...);
 				}

